server/balancer.ts

```diff
@@ -76,6 +76,7 @@ class BalancerManager {
 			payload: {
 				port: conf.get("port"),
 				region: conf.get("balancing.region"),
+				id: uuidv4(),
```

Comment:
This still makes the id random for every balancer it connects to. It needs to be constant across reconnects. It's ok if the id is not constant across service restarts (eg don't store it in a file, it being a global is fine).